### PR TITLE
Implement Recipe API

### PR DESCRIPTION
**Pull request by Faithcaio**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/1582)|[SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/1395)|*SpongeForge*

Register recipes in forge.

### CODE REVIEW

File: src/main/java/org/spongepowered/mod/mixin/core/common/MixinSpongeImplHooks.java [hunk 1]
```diff
@@ -286,4 +293,23 @@ public static void onTileEntityError(TileEntity tileEntity, CrashReport crashRep
     public static void blockExploded(Block block, World world, BlockPos blockpos, Explosion explosion) {
         block.onBlockExploded(world, blockpos, explosion);
     }
+
+    // Crafting
+
+    @Overwrite
```
Comment (stephan-gh) [inline src/main/java/org/spongepowered/mod/mixin/core/common/MixinSpongeImplHooks.java:299]: Please add a Javadoc comment with `@author` and `@reason` for both overwrites (Mixin will print compile warnings if they're missing)

**Review by stephan-gh — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/mod/mixin/core/common/MixinSpongeImplHooks.java (hunk 1 above)
Comment (Faithcaio) [reply src/main/java/org/spongepowered/mod/mixin/core/common/MixinSpongeImplHooks.java:299]: I can do that but fyi no overwrite in this class has the comments

Comment (stephan-gh) [reply src/main/java/org/spongepowered/mod/mixin/core/common/MixinSpongeImplHooks.java:299]: Oh, this is `SpongeImplHooks`... Nevermind, the purpose of all the overwrites in this class is clear, so I don't think we need them in this case.

**Comment by Faithcaio** [comment]
https://github.com/SpongePowered/SpongeCommon/commit/2a7a3664f2acbb92c1a5750ad288b1e242986e58

**Pull request closed without merging**